### ISSUE TITLE
issuer URIs with a path part don't work

### ISSUE THREAD

**Issue by zenomt** [solid/node-solid-server#1265]
i originally opened this as https://github.com/solid/oidc-auth-manager/issues/44 but perhaps i should have started here to target the right audience.

oidc-auth-mananger seems to make an invalid assumption that all OIDC Issuers are URIs with an empty path part. if a webid has an OIDC Issuer URI with a path part (like my webid https://zenomt.zenomt.com/card.ttl#me with issuer https://zenomt.com/oidc/), it can never be matched against the id_token `iss` claim because the discovered issuer URI is reduced to its origin before being compared.  see [these lines in preferred-provider.js](https://github.com/solid/oidc-auth-manager/blob/5827449e0d1d287668fff667661003e529eb8f9d/src/preferred-provider.js#L70-L73).

OIDC Issuers are [allowed to have path parts](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest).

**Comment by jaxoncreed** [comment]
Good point. This should be fixed, though until we encounter a solid-compatible openid-provider this will be a low priority.

**Comment by zenomt** [comment]
@jaxoncreed [my own OpenID provider](https://github.com/zenomt/python-webid-oidc) and my own webid would be "Solid-compatible" if this issue didn't exist.
